src/main/java/pro/cloudnode/smp/nations/listeners/PlayerConnectionListener.java

```diff
@@ -0,0 +1,38 @@
+package pro.cloudnode.smp.nations.listeners;
+
+
+import org.bukkit.entity.Player;
+import org.bukkit.event.EventHandler;
+import org.bukkit.event.Listener;
+import org.bukkit.event.player.PlayerJoinEvent;
+import org.bukkit.event.player.PlayerQuitEvent;
+import pro.cloudnode.smp.nations.locale.Messages;
+import pro.cloudnode.smp.nations.util.Nation;
+import pro.cloudnode.smp.nations.util.NationManager;
+
+import static pro.cloudnode.smp.nations.Nations.t;
+
+public class PlayerConnectionListener implements Listener {
+    @EventHandler
+    public void onPlayerJoin(PlayerJoinEvent event) {
+        Player player = event.getPlayer();
+        NationManager.updatePlayersDisplayname(player.getUniqueId());
+
+        Nation n = NationManager.getPlayerNation(player.getUniqueId());
+        // set join message
+        if (n != null) {
+            event.joinMessage(t(Messages.PLAYER_JOIN_SERVER, player, n));
+        }
```

Comment:
why change the join message? imo should just send "player is now online" or sth like that to online nation members